cmd/git-sync/main.go

```diff
@@ -82,6 +82,8 @@ var flCookieFile = flag.Bool("cookie-file", envBool("GIT_COOKIE_FILE", false),
 var flGitCmd = flag.String("git", envString("GIT_SYNC_GIT", "git"),
 	"the git command to run (subject to PATH search)")
 
+var flHTTPEnable = flag.Bool("http-enable", envBool("GIT_SYNC_HTTP_ENABLE", false),
```

Comment:
Why make this an enable flag and an address flag?  Less is more when it comes to flags.  Having the address flag also mean "enable this" seems reasonable to me - convince me otherwise?

Reply:
I suggested this to have a global feature gate for http as this is completely new behaviour.
That enables use to enable various http features by default without enabling the whole http server by default.
That was the main reasoning.

Reply:
IMO the address flag is sufficient, but it depends on what the long-term plans are. If the plan is to have this eventually be default on then having the separate flag which then gets removed makes sense.

Reply:
@stp-ip I don't quite follow.  How could you enable pprof or healthz or metrics without enabling HTTP?  My thinking was that the http-bind-address is the flag that disables all HTTP (if empty) or enables it (if not empty).  Each use-case of HTTP may want its own flag to enable/disable handler registration.  That saves us one flag, which seems small but isn't :)

Reply:
http-enable is a global flag. It needs to be enabled for all http features to function or else they are feature gated.

So pprof, healthz and metrics can be enabled by default, but http-enable is not.

The idea behind it would be to have metrics and pprof on different ports later on, if that becomes necessary and also making it easier to be backwards compatible while enabling new http enabled things by default on a case by case basis.

Happy to simplify here, just felt like a saner version to introduce this considering people also use this as a cli so disableing http by default cuts down on their need to disable http features, while cloud native people usually just need to enable http and get pprof, metrics etc.

Might just be me liking being explicit.

Reply:
The one odd thing about having the 2 flags is that you can have it `enabled` but still have an empty listen port which is just a startup error.

Reply:
This is still part of the commit series?

Reply:
I still think one flag is simpler.